ReactWindows/ReactNative/Views/Picker/ReactPickerManager.cs

```diff
@@ -0,0 +1,202 @@
+﻿using Newtonsoft.Json.Linq;
+using ReactNative.UIManager;
+using ReactNative.UIManager.Events;
+using System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace ReactNative.Views.Picker
+{
+    /// <summary>
+    /// A view manager responsible for rendering picker.
+    /// </summary>
+    public class ReactPickerManager : BaseViewManager<ComboBox, ReactPickerShadowNode>
+    {
+        private int _selected;
+
+        /// <summary>
+        /// The name of the view manager.
+        /// </summary>
+        public override string Name
+        {
+            get
+            {
+                return "RCTPicker";
+            }
+        }
+
+        /// <summary>
+        /// Sets whether a picker is enabled.
+        /// </summary>
+        /// <param name="view">a combobox view.</param>
+        /// <param name="enabled">
+        /// Set to <code>true</code> if the picker should be enabled,
+        /// otherwise, set to <code>false</code>.
+        /// </param>
+        [ReactProperty("enabled")]
+        public void SetEnabled(ComboBox view, bool enabled)
+        {
+            view.IsEnabled = enabled;
+        }
+
+        /// <summary>
+        /// Sets the selected item.
+        /// </summary>
+        /// <param name="view">a combobox instance.</param>
+        /// <param name="selected">The selected item.</param>
+        [ReactProperty("selected")]
+        public void SetSelected(ComboBox view, int selected)
+        {
+            // Temporarily disable selection changed event handler.
+            view.SelectionChanged -= OnSelectionChanged;
+
+            _selected = selected;
+            view.SelectedIndex = view.Items.Count > _selected ? _selected : -1;
+
+            if (view.SelectedIndex != -1)
+            {
+                view.Foreground = ((ComboBoxItem)(view.Items[view.SelectedIndex])).Foreground;
+            }
+
+            view.SelectionChanged += OnSelectionChanged;
+        }
+
+        /// <summary>
+        /// Populates a <see cref="ComboBox"/>
+        /// </summary>
+        /// <param name="view">a combobox instance.</param>
+        /// <param name="items">The picker items.</param>
+        [ReactProperty("items")]
+        public void SetItems(ComboBox view, JArray items)
+        {
+            // Temporarily disable selection changed event handler.
+            view.SelectionChanged -= OnSelectionChanged;
+
+            for (var index = 0; index < items.Count; index++)
+            {
+                JToken label;
+                if ((items[index] as JObject).TryGetValue("label", out label))

```

Comment:
Basically, I'd rather see the InvalidOperationException from Json.NET than the NullReferenceException.  

---
In reply to: [56703283](https://github.com/CatalystCode/react-native/pull/265#discussion_r56703283) [](ancestors = 56703283)